backend-lib/src/main/rust/lib.rs

```diff
@@ -3474,13 +3494,17 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_model_TorWalletClient_upd
                 // - v5 and above transactions ignore the argument, and parse the correct
                 //   value from their encoding.
                 let tx = Transaction::read(&tx_bytes[..], BranchId::Sapling)?;
+                found = Some(address);
 
                 decrypt_and_store_transaction(&network, &mut db_data, &tx, mined_height)
                     .map_err(|e| anyhow!("Error while decrypting transaction: {}", e))
             },
-        )
+        )?;
+
+        Ok(encode_address_check_result(env, &network, found)?.into_raw())
     });
-    unwrap_exc_or(&mut env, res, ())
+
+    unwrap_exc_or(&mut env, res, ptr::null_mut())
```

Comment:
```suggestion
    unwrap_exc_or(&mut env, res, JNI_FALSE)
```
(Here it doesn't matter as we're dropping the result and throwing the exception.)